src/pywink/devices/cloud_clock.py

```diff
@@ -0,0 +1,289 @@
+            response = self.api_interface.set_device_state(self, values)
+        return response
+
+    def get_time_dial(self):
+        for dial in self.json_state.get("dials", {}):
+            if dial['channel_configuration']['channel_id'] == "1":
+                if dial['name'] == "Time":
+                    return dial
+                return None
+        return None
+
+
+class WinkCloudClockAlarm(WinkDevice):
+    """
+    Represents a Quirky Nimbus alarm.
+    """
+
+    def __init__(self, device_state_as_json, api_interface):
+        super().__init__(device_state_as_json, api_interface)
+        self.parent = None
+        self.start_time, self.days = _parse_ical(device_state_as_json.get('recurrence'))
+
+    def state(self):
+        return self.json_state['next_at']
+
+    def set_parent(self, parent):
+        self.parent = parent
+
+    def available(self):
+        enabled = self.json_state['enabled']
+        clock = self.parent.get_time_dial()
+        return bool(enabled and clock is not None)
+
+    def set_enabled(self, enabled):
+        self.api_interface.set_device_state(self, {"enabled": enabled})
+
+    def update_state(self):
+        """ Update state with latest info from Wink API. """
+        response = self.api_interface.get_device_state(self, id_override=self.parent.object_id(),
+                                                       type_override=self.parent.object_type())
+        self._update_state_from_response(response)
+
+    def _update_state_from_response(self, response_json):
+        """
+        :param response_json: the json obj returned from query
+        :return:
+        """
+        if 'data' in response_json and response_json['data']['object_type'] == "cloud_clock":
+            cloud_clock = response_json.get('data')
+            if cloud_clock is None:
+                return False
+
+            alarms = cloud_clock.get('alarms')
+            for alarm in alarms:
+                if alarm.get('object_id') == self.object_id():
+                    self.json_state = alarm
+                    return True
+            return False
+        if 'data' in response_json:
+            alarm = response_json.get('data')
+            self.json_state = alarm
+            return True
+        self.json_state = response_json
+        return True
+
+    def set_recurrence(self, date, days=None):
+        """
+
+        :param date: Datetime object time to start/repeat
+        :param days: days to repeat (Defaults to one time alarm)
+        :return:
+        """
+        if self.parent.get_time_dial() is None:
+            _LOGGER.error("Not setting alarm, no time dial.")
+            return False
+        timezone_string = self.parent.get_time_dial()["channel_configuration"]["timezone"]
+        valid_days = ["SU", "MO", "TU", "WE", "TH", "FR", "SA"]
+        ical_string = DTSTART + timezone_string + ":" + date.strftime("%Y%m%dT%H%M%S")
+        if days is not None:
+            if days == "DAILY":

```

Comment:
I think lines 117-129 should be extracted into a helper function like `_get_ical_string`

Reply:
Yeah I have local changes I haven't pushed for alarms and this was something I did. Having trouble getting alarms to work even with the official app so not sure if the changes are working.